### DISCUSSION TITLE
Halloween features

### DISCUSSION THREAD

**Discussion by IceWaffles** [PurpurMC/Purpur#1090] [Feature Requests]
A implementation of https://www.spigotmc.org/resources/darkernights.87814/ would be a welcome addition to the existing halloween options. Who wouldn't want darker and scarier nights during halloween?

**Comment by BillyGalbreath** [comment]
Hmm. My first impression was that it just set the client's brightness level to moody (not sure how that would even be possible, but that was my thought). Then i read this comment

![image](https://user-images.githubusercontent.com/332527/182829745-70a7b98b-b7d7-462d-847c-92cfb6613237.png)

So, I'm not sure how Intelli accomplished this goal. My next thought is just a wild guess, but maybe its something in the chunk data itself. And if that's the case, could easily be replicated in a datapack.

**Reply by MelnCat** [reply]
I'm guessing that plugin changes the biome colors to make it appear darker

**Comment by IceWaffles** [comment]
Wouldn't datapacks have some sort of performance impact? Nevertheless, thanks for your answer - I'll figure out a datapack. Just asked it would be super cool if it was added to purpur though :P

**Reply by BillyGalbreath** [reply]
Contrary to popular belief, datapacks are not slow. Command functions are slow, which get heavily abused in many datapacks. This had lead to the misconception that datapacks are slow.

Datapacks for adjusting static things such as this are what they were designed for, and are the preferred way to do such things.

**Reply by IceWaffles** [reply]
> The plugin modifies client-side sky light levels to be darker during the night. The current behaviour in vanilla just keeps sky lighting at max brightness during both days and nights, which could be considered a bit of a regression from older versions of the game.

If you're interested, that's how it was achieved.
